Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
+  if (mode != kModeExec)
```

Comment:
There is no SoftwareBreakpointManager superclass, "super" here is BreakpointManager, and I can't put it there, or it will break hardware breakpoints. But I can change it to an assert.